### PR TITLE
SourceGen fixes

### PR DESCRIPTION
**Pull request by Nirmal4G**
Visual Studio (IDE/Code) project system do not know about generators yet. So, the project system complains about missing types/members.

_One solution is to trick the project system into thinking that we have the sources!_

We write the generated file to the project cone and add it to `@(Compile)` at [DesignTime](https://github.com/dotnet/project-system/blob/master/docs/design-time-builds.md) so that the project system won't complain about missing types/members
Keep in mind, that, this is a workaround, and that, we have to build the project at least once to generate the sources.

### CODE REVIEW

**Comment by Nirmal4G** [comment]
[CI for commit `Fix CodeGen issues`](https://github.com/Nirmal4G/minsk/runs/648090111)

File: src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs [hunk 1]
```diff
@@ -24,10 +24,11 @@ public void Execute(SourceGeneratorContext context)
 
             var compilation = (CSharpCompilation)context.Compilation;
 
-            var types = GetAllTypes(compilation.Assembly);
             var immutableArrayType = compilation.GetTypeByMetadataName("System.Collections.Immutable.ImmutableArray`1");
             var separatedSyntaxListType = compilation.GetTypeByMetadataName("Minsk.CodeAnalysis.Syntax.SeparatedSyntaxList`1");
             var syntaxNodeType = compilation.GetTypeByMetadataName("Minsk.CodeAnalysis.Syntax.SyntaxNode");
+
+            var types = GetAllTypes(compilation.Assembly);
```
Comment (Nirmal4G) [inline src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs:31]: Code move was intentional. Near to where it's actually used.

File: src/Minsk/Minsk.csproj [hunk 2]
```diff
@@ -4,6 +4,19 @@
     <TargetFramework>netstandard2.0</TargetFramework>
   </PropertyGroup>
 
+  <!--
+    HACK: Partially Fixes DesignTime Builds. See terrajobst/minsk#121
+  -->
+
+  <PropertyGroup>
+    <GeneratedSources>CodeAnalysis/**/*.g.cs</GeneratedSources>
+    <DefaultItemExcludes>$(GeneratedSources)</DefaultItemExcludes>
```
Comment (terrajobst) [inline src/Minsk/Minsk.csproj:13]: Is `DefaultItemExcludes` empty by default? Seems like we should be doing this:

```suggestion
    <DefaultItemExcludes>$(DefaultItemExcludes);$(GeneratedSources)</DefaultItemExcludes>
```

File: src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs [hunk 3]
```diff
@@ -90,38 +91,30 @@ public void Execute(SourceGeneratorContext context)
                 indentedTextWriter.WriteLine("}");
 
                 indentedTextWriter.Flush();
+                stringWriter.Flush();
```
Comment (terrajobst) [inline src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs:94]: Why is this needed?

File: src/Minsk/Minsk.csproj (hunk 2 above)
Comment (Nirmal4G) [inline src/Minsk/Minsk.csproj:13]: Don't need to prepend the property

File: src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs (hunk 3 above)
Comment (Nirmal4G) [inline src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs:94]: Don't we need to flush?

File: src/Minsk/Minsk.csproj (hunk 2 above)
Comment (Nirmal4G) [reply src/Minsk/Minsk.csproj:13]: Don't need to prepend the property.

File: src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs (hunk 3 above)
Comment (Nirmal4G) [reply src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs:94]: Don't we need to flush before `.ToString()`?

Comment (terrajobst) [reply src/Minsk.Generators/SyntaxNodeGetChildrenGenerator.cs:94]: I don't think so, because they are chained but it also doesn't hurt.

**Comment by terrajobst** [comment]
@Nirmal4G mind resolving the merge conflicts? This should go into the episode21 branch

**Comment by Nirmal4G** [comment]
Sure

File: azure-pipelines.yml [hunk 4]
```diff
@@ -1,47 +1,24 @@
...
-#    version: 3.1.x
-#    packageType: runtime
 - task: DotNetCoreCLI@2
-  displayName: 'dotnet build $(buildConfiguration)'
+  displayName: Build minsk ($(buildConfiguration))
   inputs:
     command: build
     projects: $(sln)
-    arguments: --configuration $(buildConfiguration)
+    arguments: -c $(buildConfiguration)
```
Comment (couven92) [inline azure-pipelines.yml:17]: Isn't `--configuration` more readable? I know `-c` is the short form, but in CI templates (where I don't have to type out the command), I prefer the long names..

*Same for line 23.*

Comment (Nirmal4G) [reply azure-pipelines.yml:17]: True but we specified `$(buildConfiguration)` that tells us what the switch's meaning is. That's why I've simplified it.
